src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCountWriter.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Writes {@link AllelicCount} instances to a tab separated table file.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public class AllelicCountWriter extends TableWriter<AllelicCount> {
+
+    private final AllelicCountTableColumns.AllelicCountTableVerbosity verbosity;
+
+    public AllelicCountWriter(final File file, final AllelicCountTableColumns.AllelicCountTableVerbosity verbosity)
+            throws IOException {
+        super(file, new TableColumnCollection(AllelicCountTableColumns.getColumns(verbosity)));
+        this.verbosity = verbosity;
+    }
+
+    @Override
+    protected void composeLine(final AllelicCount record, final DataLine dataLine) {
+        switch (verbosity) {
+            case BASIC:
+                composeLineBasic(record, dataLine);
+                break;
+            case INTERMEDIATE:
+                composeLineIntermediate(record, dataLine);
+                break;
+            case FULL:
+                composeLineFull(record, dataLine);
+                break;
+            default:
+                throw new UserException.BadInput("The AllelicCount verbosity is invalid.");
```

Comment:
You must fail early. This error should be throw an construction. and not here; Also we know that at this point this exception would not blamed to the user but rather two ourselves as it would rather indicate the presence of a bug.